src/__tests__/canonicalFields.int.test.ts

```diff
@@ -62,4 +65,111 @@ describe('/canonicalFields', () => {
       });
     });
   });
+
+  describe('POST /', () => {
+    it('creates exactly one canonical field', async () => {
+      const before: Result<{ count: number; maxId: number; now: Date }> = await db.query(`
+        SELECT COUNT(id) AS "count",
+          MAX(id) AS "maxId",
+          NOW() as "now"
+        FROM canonical_fields;
+      `);
+      logger.debug('before: %o', before);
+      const result = await agent
+        .post('/canonicalFields')
+        .type('application/json')
+        .send({
+          label: '🏷️',
+          shortCode: '🩳',
+          dataType: '📊',
+        })
+        .expect(201);
+      const after: Result<{ count: number; maxId: number; now: Date }> = await db.query(`
+        SELECT COUNT(id) AS "count",
+          MAX(id) AS "maxId",
+          NOW() as "now"
+        FROM canonical_fields;
+      `);
+      logger.debug('after: %o', after);
+      expect(before.rows[0].count).toEqual('0');
+      expect(result.body).toMatchObject({
+        id: expect.any(Number) as number,
+        label: '🏷️',
+        shortCode: '🩳',
+        dataType: '📊',
+        createdAt: expect.stringMatching(isoTimestampPattern) as string,
```

Comment:
This and the above `id` expectation are nice and flexible. If we expect 0 rows before, like on line 94, then we should be able to expect that the first row inserted should be 1, right? Unless the db has not been created from scratch. Anyway, this is probably fine, good enough. But we can insert the exact datetime and then verify that we get the same on the other side.

Minor point regarding the timestamp pattern: it's ISO but more specifically an IETF RFC 3339 timestamp pattern.

Reply:
Both great points -- I'll make an issue to clean that (and other things you've flagged) up